### ISSUE TITLE
Bug: serd_reader_read_chunk does not support NQuads

### ISSUE THREAD

**Issue by mcb5637** [drobilla/serd#42]
The following code does not properly read NQuads (Graph is always null):
```cpp
auto* r = serd_reader_new(...);
serd_reader_start_source_stream(...);
while (serd_reader_read_chunk(...) ...) {...};
```

(Reading a file directly via `serd_reader_read_file_handle` works, but reading via a simple adaptor from a c++ `std::istream` to `serd_reader_read_chunk` does not)

Possible cause: 
`serd_reader_read_chunk` calls `read_statement` which calls `read_n3_statement` direcly, without checking the readers syntax mode.

**Comment by drobilla** [comment]
Fixed in 4196ab792, thanks.

Please let me know if this is working as expected in your situation.

**Comment by mcb5637** [comment]
Yes, does work now.
Thanks for the fast fix.
